ironfish-cli/src/commands/wallet/multisig/dkg/create.ts

```diff
@@ -2,102 +2,228 @@
+
+    this.log('\nRound 2 Public Package:')
+    this.log(round2Result.publicPackage)
+    this.log('\n============================================')
+    this.log('\nShare your Round 2 Public Package with other participants.')
+
     await this.retryStep(async () => {
       await this.performRound3(
         client,
-        name,
+        accountName,
+        participantName,
         round2Result,
         round1PublicPackages,
         totalParticipants,
+        ledger,
       )
     })
 
     this.log('Multisig account created successfully using DKG!')
   }
 
-  private async getNameForMultisigAccount(client: RpcClient): Promise<string> {
-    const name = await ui.inputPrompt(
-      'Enter a name for your account and participant identity',
-      true,
-    )
-
+  private async getParticipant(client: RpcClient, participantName?: string) {
     const identities = (await client.wallet.multisig.getIdentities()).content.identities
 
-    const accounts = (await client.wallet.getAccounts()).content.accounts
-
-    const foundAccount = accounts.find((account) => account === name)
+    if (participantName) {
+      const foundIdentity = identities.find((i) => i.name === participantName)
+      if (!foundIdentity) {
+        throw new Error(`Participant with name ${participantName} not found`)
+      }
 
-    if (foundAccount) {
-      throw new Error(`Account with name ${name} already exists`)
+      return {
+        name: foundIdentity.name,
+        identity: foundIdentity.identity,
+      }
     }
 
-    const foundIdentity = identities.find((identity) => identity.name === name)
+    const name = await ui.inputPrompt('Enter the name of the participant', true)
+    const foundIdentity = identities.find((i) => i.name === name)
 
     if (foundIdentity) {
-      throw new Error(`Identity with name ${name} already exists`)
+      this.log('Found an identity with the same name')
+
+      return {
+        ...foundIdentity,
+      }
+    }
+
+    const identity = (await client.wallet.multisig.createParticipant({ name })).content.identity
+
+    return {
+      name,
+      identity,
+    }
+  }
+
+  private async getAccountName(client: RpcClient, accountName?: string) {
+    let name: string
+    if (accountName) {
+      name = accountName
+    } else {
+      name = await ui.inputPrompt('Enter a name for the new multisig account', true)
+    }
+
+    const accounts = (await client.wallet.getAccounts()).content.accounts
+
+    if (accounts.find((a) => a === name)) {
+      this.log('An account with the same name already exists')
+      name = await ui.inputPrompt('Enter a new name for the account', true)
     }
 
     return name
   }
 
+  async getIdentityFromLedger(
+    ledger: Ledger,
+    client: RpcClient,
+    name?: string,
+  ): Promise<{
+    name: string
+    identity: string
+  }> {
+    // TODO(hughy): support multiple identities using index
+    const identity = await ledger.dkgGetIdentity(0)
```

Comment:
nit - maybe we should log that we're requesting the identity? We ask the user to approve the request, but it might not be clear what the request is

Reply:
done, changed the prompt message